### PR TITLE
Expand explanation for compatibility mouse events

### LINKED ISSUE #454: Order of pointerover/enter/move and corresponding mouse events is different on browsers
`data:text/html,<input id=input><script>input.onpointerenter = input.onmouseenter = input.onpointerleave = input.onmouseleave = input.onpointerover = input.onmouseover = input.onpointerout = input.onmouseout = input.onpointermove = input.onmousemove = ev =>console.log(ev.type)</script>`

When hovering over the input, the console output is:

Chrome: pointerover->pointerenter->mouseover->mouseenter->pointermove->mousemove->pointerout->pointerleave->mouseout->mouseleave
Firefox: pointerover->pointerenter->pointermove->mouseover->mouseenter->mousemove->pointerout->pointerleave->mouseout->mouseleave
Safari: pointerover->mouseover->pointerenter->mouseenter->pointermove->mousemove->pointerout->mouseout->pointerleave->mouseleave

Somehow the timing that pointermove and mousemove appear is different.

* The pointer/mousemove happens immediately on pointerenter on all browsers
* Events are somehow grouped on Firefox and Chrome (pointerover+pointerenter always happen together without a compatibility event between it, for example)
* And the grouping behaviors are different.

I'm not quite sure this grouping is specced. https://w3c.github.io/pointerevents/#mapping-for-devices-that-support-hover looks like it doesn't even give the mapping for over/enter/out/leave.

### PR DESCRIPTION
**Pull request by patrickhlauke**
* Add note about high level "interleaved" events, and the reality of what user agents may do, especially on touchscreens, as long as the relative order is consistent

Closes #454

***
<a href="https://pr-preview.s3.amazonaws.com/w3c/pointerevents/pull/459.html" title="Last updated on Oct 26, 2022, 8:29 PM UTC (a1adb2d)">Preview</a> | <a href="https://pr-preview.s3.amazonaws.com/w3c/pointerevents/459/1074161...a1adb2d.html" title="Last updated on Oct 26, 2022, 8:29 PM UTC (a1adb2d)">Diff</a>

### CODE REVIEW

**Comment by patrickhlauke** [comment]
I think this PR addresses the actual question of #454 (basically, that the exact sequence of when a compat mouse event fires in relation to a pointer event is undefined and left up to UAs, as long as the overall relative order reflects a consistent legacy mouse pointer).

https://github.com/w3c/pointerevents/pull/458 tacles a slightly separate, but related, issue (since the original question of #454 was not about touch, but mouse). Will review/merge this separately

File: index.html [hunk 1]
```diff
@@ -1182,6 +1182,10 @@ <h2>Populating and maintaining the coalesced and predicted events lists</h2>
...
         <p>The compatibility mapping with mouse events is an OPTIONAL feature of this specification. User agents are encouraged to support the feature for best compatibility with existing legacy content.</p>
+        <div class="note">
+            <p>At a high level, compatibility mouse events are intended to be "interleaved" with their respective pointer events. However, this specific order is not mandatory, and user agents that implement compatibility mouse events MAY decide to delay or group the dispatch of mouse events, as long as their relative order is consistent.</p>
+            <p>Particularly in the case of touchscreen inputs, user agents MAY apply additional heuristics for gesture recognition (unless explicitly suppressed by authors through <a><code>touch-action</code></a>). This may result in compatibility mouse events only being dispatched after {{GlobalEventHandlers/pointerup}}, if the user agent determined that an interaction was not intended as a particular gesture. These specifics of user agent gesture recognition are not defined in this specification, and they may differ between implementations.</p>
```
Comment (mustaqahmed) [inline index.html]: Perhaps clarify further what "after `pointerup`" mean.  My suggestion is to rephrase the second sentence in this para to something like this:

> During a sequence of events between a `pointerdown` event and a `pointerup` event, the gesture recognition may have to wait until the `pointerup` event to detect or ignore a gesture, and as a result the compatibility mouse events for the whole sequence may be dispatched together after the last `pointerup` event.

**Review by mustaqahmed — CHANGES_REQUESTED** [review]
Thanks, this is looking great.  I am requesting a minor change to clarify an easy "batching" case...something along the lines of what we discussed in our group meeting few weeks ago.

File: index.html (hunk 1 above)
Comment (patrickhlauke) [reply index.html]: Tweaked slightly, but yes looking good. Thanks for the suggestion.

**Review by mustaqahmed — APPROVED** [review]
LGTM, thanks.  I will land it now.

**Pull request merged**